sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -576,6 +576,13 @@ class Analyzer(
     /** Extracts a [[Generator]] expression and any names assigned by aliases to their output. */
     private object AliasedGenerator {
       def unapply(e: Expression): Option[(Generator, Seq[String])] = e match {
+        case Alias(g: Generator, name) if g.elementTypes.size > 1 => {
```

Comment:
I think the real fix is to stop setting defaults in in the parser and instead let the analyzer do it (as we do with MultiAlias today).  However, thats a pretty big change, so I'd be okay just checking with a regex here for now.